### PR TITLE
[GLUTEN-2362][CORE] Record Substrait modifications and remove arena option

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?

Record Substrait modifications and remove arena option.

(Fixes: \#2362)

## How was this patch tested?

Testing on CI.

### CODE REVIEW

**Review by zzcclp — DISMISSED** [review]
LGTM

File: docs/developers/SubstraitModifications.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+considered.  
+
+## Modifications to algebra.proto
+
+* Added `JsonReadOptions` and `TextReadOptions` in `FileOrFiles`([#1584](https://github.com/oap-project/gluten/pull/1584)).
+* Changed join type `JOIN_TYPE_SEMI` to `JOIN_TYPE_LEFT_SEMI` and `JOIN_TYPE_RIGHT_SEMI`([#408](https://github.com/oap-project/gluten/pull/408)).
+* Added `WindowRel`, added `column_name` and `window_type` in `WindowFunction`,
+changed `Unbounded` in `WindowFunction` into `Unbounded_Preceding` and `Unbounded_Following`, and added WindowType([#485](https://github.com/oap-project/gluten/pull/485)).
+* Added `output_schema` in RelRoot([#1901](https://github.com/oap-project/gluten/pull/1901)).
+* Added `ExpandRel`([#519](https://github.com/oap-project/gluten/pull/519)).
```
Comment (JkSelf) [inline docs/developers/SubstraitModifications.md:23]: Please change the ExpandRel link to https://github.com/oap-project/gluten/pull/1361.

File: docs/developers/SubstraitModifications.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+layout: page
+title: Substrait Modifications
+nav_order: 4
+parent: Developer Overview
+---
+# Substrait Modifications in Gluten
```
Comment (winningsix) [inline docs/developers/SubstraitModifications.md:7]: Do we need to mention the base version of Substrait?

File: docs/developers/SubstraitModifications.md (hunk 1 above)
Comment (rui-mo) [reply docs/developers/SubstraitModifications.md]: Fixed.

File: docs/developers/SubstraitModifications.md (hunk 2 above)
Comment (rui-mo) [reply docs/developers/SubstraitModifications.md:7]: Added the base version.

**Review by JkSelf — APPROVED** [review]
LGTM

**Pull request merged**